### PR TITLE
Integration tests for list addresses with filtering and outstanding list addresses tests via CLI

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

#356

# Overview

- [ ] I have added integration tests for list address with filtering and added remaining cases for listing addresses via CLI

# Comments

### CODE REVIEW

File: lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs [hunk 1]
```diff
@@ -10,41 +10,205 @@ module Test.Integration.Scenario.CLI.Addresses
...
+        let filters = [ "usedd"
+                      , "uused"
+                      , "unusedd"
+                      , "uunused"
+                      , "USED"
+                      , "UNUSED"
+                      , "-1000"
+                      , "44444444"
+                      , "*"
+                      ]
```
Comment (KtorZ) [inline lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs]: Careful about indentation here

File: lib/http-bridge/test/integration/Test/Integration/Scenario/API/Addresses.hs [hunk 2]
```diff
@@ -69,26 +69,80 @@ spec = do
...
+        let filters = [ "?state=usedd"
+                      , "?state=uused"
+                      , "?state=unusedd"
+                      , "?state=uunused"
+                      , "?state=USED"
+                      , "?state=UNUSED"
+                      , "?state=-1000"
+                      , "?state=44444444"
+                      , "?state=*"
+                      ]
```
Comment (KtorZ) [inline lib/http-bridge/test/integration/Test/Integration/Scenario/API/Addresses.hs]: No variable vertical alignement please :)

File: lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs [hunk 3]
```diff
@@ -10,41 +10,205 @@ module Test.Integration.Scenario.CLI.Addresses
...
+            if (title == "40 chars hex") then
+                e `shouldBe` errMsg404NoWallet "1111111111111111111111111111111111111111\n"
+            else
+                e `shouldBe` "wallet id should be an hex-encoded string of\
+                    \ 40 characters\n"
+
+            if (title == "40 chars hex") then
+                c `shouldBe` ExitSuccess
+            else
+                c `shouldBe` ExitFailure 1
```
Comment (KtorZ) [inline lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs:178]: Quite unrelated but this is odd? Do we return a success on 404?

**Review by KtorZ — APPROVED** [review]

File: lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs (hunk 3 above)
Comment (piotr-iohk) [reply lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs:178]: Yes, I think so too, but I was told that this is expected... ;) -> https://github.com/input-output-hk/cardano-wallet/pull/278#discussion_r285063171

Comment (KtorZ) [reply lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs:178]: Then I've very poorly expressed myself. I meant that you were right and that this oddity is not correct. May you open a bug ticket to keep track of that please?

Comment (piotr-iohk) [reply lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs:178]: Ah, ok. My bad for misunderstanding then. Will open a bug :+1:

File: lib/http-bridge/test/integration/Test/Integration/Scenario/API/Addresses.hs (hunk 2 above)
Comment (piotr-iohk) [reply lib/http-bridge/test/integration/Test/Integration/Scenario/API/Addresses.hs]: Will I ever learn? :sweat_smile:  Fixed.

**Pull request merged**